core/node/storage/pg_stream_store.go

```diff
@@ -135,22 +150,37 @@ func (s *PostgresStreamStore) txRunnerWithUUIDCheck(
 	)
 }
 
-// processErrorForMissingPartition looks for the specific error type returned when a table is not found
-// and returns that as a NOT_FOUND river error for the specified stream.
-func processErrorForMissingPartition(err error, streamId StreamId) error {
-	var pgErr *pgconn.PgError
-	if errors.As(err, &pgErr) {
-		if pgErr.Code == pgerrcode.UndefinedTable {
-			return RiverError(Err_NOT_FOUND, "stream not found in local storage", "streamId", streamId)
-		}
-	}
-	return err
+// createPartitionSuffix determines the partition mapping for a particular stream id the
+// hex encoding of the first byte of the xxHash of the stream ID.
+func createPartitionSuffix(streamId StreamId, reducedParitions bool) string {
+	// Do not hash the stream bytes directly, but hash the hex encoding of the stream id, which is
+	// what we store in the database. This leaves the door open for installing xxhash on postgres
+	// and debugging this way in the future.
+	hash := xxhash.Sum64String(streamId.String())
+	// For test installations, expect 8 partitions, taking 8 bits from the hash
+	if reducedParitions {
+		bt := hash % 8
+		return fmt.Sprintf("%02x", bt)
+	}
+	return fmt.Sprintf("%016x", hash)[:2]
+}
+
+type escapeSqlOpts struct {
+	migrated bool
+	// When test are running, the migrated database schema is created with 8 partitions instead of
+	// 256. In this case, expect only 8 partitions.
+	testMode bool
 }
 
 // sqlForStream escapes references to partitioned tables to the specific partition where the stream
 // is assigned whenever they are surrounded by double curly brackets.
-func sqlForStream(sql string, streamId StreamId) string {
-	suffix := createTableSuffix(streamId)
+func sqlForStream(sql string, streamId StreamId, opts escapeSqlOpts) string {
```

Comment:
please make this func a method of PostgresStreamStore, then it will have access to test flags and there will be enough just to have `migrated bool` as arg and each call site will be more concise.